linker/shared/src/main/scala/org/scalajs/linker/standard/LinkTimeProperties.scala

```diff
@@ -38,30 +39,119 @@ private[linker] final class LinkTimeProperties (
       LinkTimeString(ScalaJSVersions.current)
   )
 
-  def validate(name: String, tpe: jstpe.Type): Boolean = {
+  def validate(name: String, tpe: Type): Boolean = {
     linkTimeProperties.get(name).exists {
-      case _: LinkTimeBoolean => tpe == jstpe.BooleanType
-      case _: LinkTimeInt     => tpe == jstpe.IntType
-      case _: LinkTimeString  => tpe == jstpe.StringType
+      case _: LinkTimeBoolean => tpe == BooleanType
+      case _: LinkTimeInt     => tpe == IntType
+      case _: LinkTimeString  => tpe == StringType
     }
   }
 
-  def transformLinkTimeProperty(prop: js.LinkTimeProperty): js.Literal = {
+  def transformLinkTimeProperty(prop: LinkTimeProperty): Literal = {
     val value = linkTimeProperties.getOrElse(prop.name,
         throw new IllegalArgumentException(s"link time property not found: '${prop.name}' of type ${prop.tpe}"))
     value match {
       case LinkTimeBoolean(value) =>
-        js.BooleanLiteral(value)(prop.pos)
+        BooleanLiteral(value)(prop.pos)
       case LinkTimeInt(value) =>
-        js.IntLiteral(value)(prop.pos)
+        IntLiteral(value)(prop.pos)
       case LinkTimeString(value) =>
-        js.StringLiteral(value)(prop.pos)
+        StringLiteral(value)(prop.pos)
     }
   }
+
+  /** Try and evaluate a link-time expression tree as a boolean value.
+   *
+   *  This method assumes that the given `tree` is valid according to the
+   *  `ClassDefChecker` and that its `tpe` is `BooleanType`.
+   *
+   *  Returns `None` if any subtree that needed evaluation was an invalid
+   *  `LinkTimeProperty` (i.e., one that does not pass the [[validate]] test).
+   */
+  def tryEvalLinkTimeBooleanExpr(tree: Tree): Option[Boolean] =
```

Comment:
I know this is consistent with the current design of `LinkTimeProperties`, but IMO this really does not belong here.

In general, in the Scala.js linker, we separate data from it's related functionality. To me, it would feel much more natural to stop at the `linkTimeProperties` map.

I feel the structure here deserves some love:
- With the introduction of desugar, this should probably be in `frontend`?
- I'm unsure whether `LinkTimeValue` is worth the abstraction overhead: it does strip literals of their position, but it also seems to introduce quite some boilerplate (we could have `linkTimeProperties: Map[String, Literal]`).

I'm not entirely sure how to best proceed here. WDYT?